### ISSUE TITLE
[PRODUCT]: CNRFC NBM QPF

### ISSUE THREAD

**Issue by brettpalmberg** [USACE/cumulus#250]
## Product Source (URL Preferred)

URL Structure: https://www.cnrfc.noaa.gov/archive/YYYY/MMM/netcdfQpfNbm/QPF.YYYYMMdd_HHMM.nc.gz

01/07/13/19Z runs of NBM, updated within about 1.5 hours of model time

Additional information from email:

> The NBM data is also available via NOMADS https://nomads.ncep.noaa.gov in GRIB2 format, and you’re already grabbing NBM data (just not sure about your source).  Either way, we need to version the data sources appropriately.  The nice thing about the CNFRC NBM data is that they reprocess the 3- and 6-hourly portions of the forecast and combine into a single hourly dataset.  For your reference, here are differences between the NOMADS and the CNRFC NBM datasets:

  | NOMADS | CNRFC
-- | -- | --
Spatial Resolution | 2.5 km | 4.7 km
Temporal Resolution ~for temps~ | Hourly, 3-hourly, 6-hourly | ~Hourly~ 6-hourly
Issuance | Issued hourly, but forecast length varies from 18 hours to 10-11 days | Every six hours, always with at least 10 days of forecast length
Format | GRIB2 | NetCDF
 
**Addition Information from Recent Email**

> NBM QPF (6 hourly):
General Link: https://www.cnrfc.noaa.gov/archive/YYYY/MMM/netcdfQpfNbm/QPF.YYYYMMdd_HHMM.nc.gz
Example link: https://www.cnrfc.noaa.gov/archive/2022/Aug/netcdfQpfNbm/QPF.20220822_0700.nc.gz
Index: Not found
Comment: Able to download at 0100, 0700, 1300, 1900, as expected

Availability is at 0100, 0700, 1300, 1900. (cc @jeffsuperglide, timestamps related to finding sample files)

## Product Format

- [ ] Geotiff
- [ ] Grib2
- [x] NetCDF
- [ ] Other (Please provide description below)

**Comment by brettpalmberg** [comment]
Updated description with most recent information, under heading `Addition Information from Recent Email`

**Comment by brettpalmberg** [comment]
Recent sample files from December 08, 2022:

- https://www.cnrfc.noaa.gov/archive/2022/Dec/netcdfQpfNbm/QPF.20221208_0100.nc.gz
- https://www.cnrfc.noaa.gov/archive/2022/Dec/netcdfQpfNbm/QPF.20221208_0700.nc.gz
- - https://www.cnrfc.noaa.gov/archive/2022/Dec/netcdfQpfNbm/QPF.20221208_1300.nc.gz
- - https://www.cnrfc.noaa.gov/archive/2022/Dec/netcdfQpfNbm/QPF.20221208_1900.nc.gz

Note: File posting time unknown for Airflow DAG data retrieval. Will determine based on trial and error and dial-in based on results.

**Comment by adamscarberry** [comment]
DB migration complete and merged into develop.  Working on airflow dag now.

cc @jeffsuperglide

**Comment by adamscarberry** [comment]
NBM QPF is in inches, migration needs fixed by @adamscarberry

**Comment by brettpalmberg** [comment]
Grids from CNRFC reference a Polar Stereographic coordinate reference system (CRS). The CRS is not specified in the source files in a manner that can be recognized by [GDAL utilities](https://gdal.org/index.html).

We implemented a coordinate reference system (CRS) approach in USACE/cumulus-geoproc that assigns the geotransform and projection to the grid file to the best of our knowledge so it can be recognized by GDAL. This would benefit from coordination and review from the data provider (CNRFC) on the approach.

Data acquisition and processing is enabled in develop & stable versions of the cumulus app. A tag was added to the CNRFC products "BETA Testing" to inform users that this product is still under testing/verification. Closing issue now.

cc @FHanbali @adamscarberry @jeffsuperglide
